packages/agreement-process/test/testUpgradeAgreement.ts

```diff
@@ -0,0 +1,1273 @@
+/* eslint-disable sonarjs/cognitive-complexity */
+/* eslint-disable fp/no-delete */
+/* eslint-disable functional/immutable-data */
+import { fail } from "assert";
+import { generateMock } from "@anatine/zod-mock";
+import {
+  getMockAgreement,
+  getMockCertifiedTenantAttribute,
+  getMockDeclaredTenantAttribute,
+  getMockDescriptorPublished,
+  getMockEService,
+  getMockEServiceAttribute,
+  getMockTenant,
+  getMockVerifiedTenantAttribute,
+  getRandomAuthData,
+  StoredEvent,
+} from "pagopa-interop-commons-test";
+import {
+  Agreement,
+  AgreementAddedV1,
+  AgreementConsumerDocumentAddedV1,
+  AgreementDocumentId,
+  AgreementId,
+  AgreementV1,
+  AttributeId,
+  Descriptor,
+  DescriptorId,
+  Document,
+  EService,
+  EServiceId,
+  StampsV1,
+  TenantId,
+  agreementState,
+  descriptorState,
+  generateId,
+  protobufDecoder,
+  unsafeBrandId,
+} from "pagopa-interop-models";
+import { afterAll, beforeAll, describe, expect, it, vi } from "vitest";
+import { v4 as uuidv4 } from "uuid";
+import { FileManagerError, userRoles } from "pagopa-interop-commons";
+import {
+  toAgreementStateV1,
+  toAgreementV1,
+  toStampV1,
+} from "../src/model/domain/toEvent.js";
+import { config } from "../src/utilities/config.js";
+import {
+  agreementAlreadyExists,
+  agreementNotFound,
+  agreementNotInExpectedState,
+  descriptorNotFound,
+  eServiceNotFound,
+  missingCertifiedAttributesError,
+  noNewerDescriptor,
+  operationNotAllowed,
+  publishedDescriptorNotFound,
+  tenantIdNotFound,
+  unexpectedVersionFormat,
+} from "../src/model/domain/errors.js";
+import {
+  addOneAgreement,
+  addOneEService,
+  addOneTenant,
+  readAgreementEventByVersion,
+  readLastAgreementEvent,
+} from "./utils.js";
+import {
+  eservices,
+  tenants,
+  postgresDB,
+  agreementService,
+  agreements,
+  fileManager,
+} from "./agreementService.integration.test.js";
+
+export const testUpgradeAgreement = (): ReturnType<typeof describe> =>
+  describe("Upgrade Agreement", () => {
+    const TEST_EXECUTION_DATE = new Date();
```

Comment:
This is probably not needed, if you just use `new Date()` in vi.setSystemTime and then again `new Date()` around the rest of the test (in the other tests I wrote it works at least 😛 )

Reply:
Sure is not necessary 🍃  but I would to leave it as is for a pure readability, during assertion could be more convenient reading a const because intends some logic on date during the test,  the `new Date()` seems only a mocked value.

Reply:
👍